pkg/kubernetes/scripts/virtual-machines/types.jsx

```diff
@@ -105,3 +108,47 @@ export type PersistenVolume = {
 }
 
 export type PersistenVolumes = Array<PersistenVolume>;
+
+export type PodMetadata = {
+    "name": string,
+    "generateName": ?string,
+    "namespace": string,
+    "selfLink": string,
+    "uid": string,
+    "resourceVersion": string,
+    "creationTimestamp": string,
+    "labels": Labels,
+    "annotations": Annotations
+};
+
+export type PodVolume = {
+    "name": string,
+    "hostPath": Object
+};
+
+export type PodSpec = {
+    "volumes": Array<PodVolume>,
+    "containers": Array<Object>,
+    "restartPolicy": string,
+    "terminationGracePeriodSeconds": number,
+    "dnsPolicy": string,
+    "serviceAccountName": string,
+    "serviceAccount": string,
+    "nodeName": string,
+    "hostPID": boolean,
+    "securityContext": Object,
+    "imagePullSecrets": Array<Object>,
+    "schedulerName": string
+};
+
+export type Pod = {
+    "kind": "Pod",
+    "apiVersion": string,
+    "metadata": PodMetadata,
+    "spec": PodSpec,
+    "status": ?{
+        "phase": ?string
+    }
+};
```

Comment:
This specifies a lot of Pod fields that are currently unused. Up to you, but how stable will this be in the future?

Reply:
Most of it should probably stay stable on this level. But you are right, it's recently not used, so reducing the level of detail to just the high-level structure.